sdk/src/main/java/com/google/cloud/dataflow/sdk/util/common/Counter.java

```diff
@@ -241,6 +246,35 @@ public AggregationKind getKind() {
    */
   public abstract T getAggregate();
 
+  /** Returns the naming context associated with this counter. */
+  public NameContext getNameContext() {
+    return this.name.getContext();
+  }
+
+  /** Sets the naming context associated with this counter. */
+  public void setNameContext(NameContext context, NameType type) {
```

Comment:
Can you expand the Javadoc here?

`@Nullable` on context, and some explanation of the behavior? It has some weird requirements (if type == UNSTRUCTURED, and context != null then it throws an error).